src/Idris/IBC.hs

```diff
@@ -113,15 +115,60 @@ loadPkgIndex pkg = do ddir <- runIO $ getIdrisLibDir
                       fp <- findPkgIndex pkg
                       loadIBC True fp
 
-bencode :: Binary a => FilePath -> a -> IO ()
-bencode f d = B.writeFile f (compress (encode d))
 
-bdecode :: Binary b => FilePath -> IO b
-bdecode f = do d' <- B.readFile f
-               either
-                 (\(_, e) -> error $ "Invalid / corrupted zip format on " ++ show f ++ ": " ++ e)
-                 (return . decode)
-                 (decompressEither d')
+entries :: IBCFile -> Idris [Entry]
+entries i = do
```

Comment:
Why does this function need to be in the Idris monad? It looks pure to me.

Also, I think it would be cleaner to drop the "do" and just have the expression in the body, eg

```
entries :: IBCFile -> [Entry]
entries = [ toEntry....
```
or
```
entries :: IBCFile -> Idris [Entry]
entries = return [ toEntry....
```
for the monadic version.